server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/RangeAggregator.java

```diff
@@ -640,18 +654,33 @@ private abstract static class NumericRangeAggregator extends RangeAggregator {
         @Override
         public LeafBucketCollector getLeafCollector(LeafReaderContext ctx, LeafBucketCollector sub) throws IOException {
             final SortedNumericDoubleValues values = ((ValuesSource.Numeric) this.valuesSource).doubleValues(ctx);
-            return new LeafBucketCollectorBase(sub, values) {
-                @Override
-                public void collect(int doc, long bucket) throws IOException {
-                    if (values.advanceExact(doc)) {
-                        final int valuesCount = values.docValueCount();
-                        for (int i = 0, lo = 0; i < valuesCount; ++i) {
-                            final double value = values.nextValue();
-                            lo = NumericRangeAggregator.this.collect(sub, doc, value, bucket, lo);
+            final NumericDoubleValues singleton = FieldData.unwrapSingleton(values);
+
+            if (singleton != null) {
+                super.singletonRanges++;
+                return new LeafBucketCollectorBase(sub, values) {
+                    @Override
+                    public void collect(int doc, long bucket) throws IOException {
+                        if (singleton.advanceExact(doc)) {
+                            NumericRangeAggregator.this.collect(sub, doc, singleton.doubleValue(), bucket, 0);
                         }
                     }
-                }
-            };
+                };
+            } else {
```

Comment:
Would you kindly drop the `else` if you are returning? I'm on a semi-irrational quest to remove indentation.